lib/resources/dstu2/medication_statement.yaml

```diff
@@ -0,0 +1,259 @@
+                 "code": {
+                   "coding": [
+                     "system": "http://hl7.org/fhir/v3/vs/GTSAbbreviation",
+                     "code": "TID",
+                     "display": "TID"],
+                   "text": "TID"}}
+
+  - name: asNeeded[x]
+    required: 'No'
+    cardinality: 0..1
+    type: boolean|CodeableConcept
+    description: Whether the Medication is only taken when needed within a specific dosing schedule.
+    example: |
+               asNeededCodeableConcept
+                 {"asNeededCodeableConcept": {
+                   "coding": [
+                     {"system": "http://snomed.info/sct",
+                      "code": "25064002",
+                      "display": "for headache"}]}}
+
+               asNeededBoolean
+                 {"asNeededBoolean" : true}
+    url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dosage.asNeeded_x_
+
+  - name: route
+    required: 'No'
+    cardinality: 0..1
+    type: CodeableConcept
+    description: >
+                     A code specifying the route or physiological path of administration of a therapeutic
+                     agent into or onto a subject.
+    example: |
+              {"route": {
+                "coding": [
+                  {"system": "http://snomed.info/sct",
+                   "code": "41974700",
+                   "display": "Chew - dosing instruction imperative (qualifier value)"}],
+                "text": "Chewed"}}
+    binding:
+      description: >
+                      A coded concept describing the route or physiological path of administration of a
+                      therapeutic agent into or onto the body of a subject.
+      terminology:
+      - display: SNOMED CT
+        system: http://snomed.info/sct
+
+  - name: quantity[x]
+    required: 'No'
+    cardinality: 0..1
+    type: SimpleQuantity|Range
+    description: The amount of therapeutic or other substance given at one administration event.
+    example: |
+               quantityQuantity
+                 {"quantityQuantity": {
+                   "value": 2,
+                   "unit": "tabs",
+                   "system": "http://unitsofmeasure.org",
+                   "code": "tbl"}}
+
+               quantityRange
+                 {"quantityRange": {
+                   "low": {
+                     "value": 2,
+                     "unit": "tabs"},
+                   "high": {
+                     "value": 3,
+                     "unit": "tabs"}}}
+    url: http://hl7.org/fhir/DSTU2/medicationstatement-definitions.html#MedicationStatement.dosage.quantity_x_
+
+- name: patient
+  required: 'Yes'
+  cardinality: 1..1
+  type: Reference (Patient)
+  description: Who is/was taking the medication.
+  example: '{"patient": {"reference": "Patient/5366327"}}'
+
+- name: reasonForUse[x]
+  required: 'No'
+  cardinality: 0..1
+  type: CodeableConcept|Reference(Condition)
```

Comment:
Additionally when fixing formatting, put a space between 'Reference' and the () as well as spaces between the items inside the parentheses if there's more than one. 

Reference (Condition)
Reference (Condition | Medication)'